ocaml/database/db_xml.ml

```diff
@@ -118,15 +123,32 @@ module From = struct
 						let ctime = match ctime_l with | [(_,ctime_s)] -> Int64.of_string ctime_s | _ -> 0L in
 						let mtime = match mtime_l with | [(_,mtime_s)] -> Int64.of_string mtime_s | _ -> 0L in
 						let row = List.fold_left (fun row ((_, k), v) -> 
-                                                        let table_schema = Schema.Database.find tblname schema.Schema.database in
-                                                        let column_schema = try
-                                                                Schema.Table.find k table_schema
-                                                        with Not_found ->
-                                                                raise (Unmarshall_error (Printf.sprintf "Unexpected column in table %s: %s" tblname k))
-                                                        in
-                                                        let value = Schema.Value.unmarshal column_schema.Schema.Column.ty (Xml_spaces.unprotect v) in
-                                                        let empty = column_schema.Schema.Column.empty in
-							Row.update mtime k empty (fun _ -> value) (Row.add ctime k value row)
+							let table_schema = Schema.Database.find tblname schema.Schema.database in
+							try
+								let column_schema = Schema.Table.find k table_schema in
+								let value = Schema.Value.unmarshal column_schema.Schema.Column.ty (Xml_spaces.unprotect v) in
+								let empty = column_schema.Schema.Column.empty in
+								Row.update mtime k empty (fun _ -> value) (Row.add ctime k value row)
+							with Not_found ->
+								(* This means there's an unexpected field, so we should normally fail. However, fields
+								 * present in Tech Preview releases are permitted to disappear on upgrade, so suppress
+								 * such errors on such upgrades. *)
+								let exc = Unmarshall_error (Printf.sprintf "Unexpected column in table %s: %s" tblname k) in
+								let (this_maj, this_min) = try
+									schema_vsn_of_manifest manifest
+								with Not_found ->
+									(* Probably the database didn't have a <manifest> at the start. So at this point
+									 * we don't know the schema version of the database we're loading. *)
+									D.error "Unmarshalling removed column %s from table %s but don't know schema version because manifest not yet read" k tblname;
+									raise exc
+								in
+								if List.exists (fun (tp_maj, tp_min) -> (tp_maj, tp_min) = (this_maj, this_min)) Datamodel.tech_preview_releases then (
```

Comment:
This `List.exists` is not necessary. A `List.mem` should suffice:
```ocaml
if List.mem (this_maj, this_min) Datamodel.tech_preview_releases ...
```

Then you don't even need to unpack into a tuple of `(this_maj, this_min)` although you probably want to keep that for the warning log message.